docs/decisions/0001-include-slug-in-uid.rst

```diff
@@ -0,0 +1,41 @@
+1. Include a slug in the UID returned by the IdP
+------------------------------------------------
+
+Status
+------
+Accepted
+
+Context
+-------
+Users that opt to authenticate using social authentication would have an association between the user in the platform
+and the id given by the IdP. In a multi tenancy context there *will* be the case that multiple sites may use the same
+Provider making unfeasible to identify site ownership from each association.
+
+Decision
+--------
+A rather simple solution was already present in the saml backend, which prepends the slug to de uid. Therefore we have
+decided to do the same for the other providers.
+
+The design
+===========
+We alter the ``get_user_id`` function from the generic OIDC backend to return a modified version of the uid that includes
+the slug in the format ``slug:uid``. The slug is obtained from the provider configuration and must be set under the
+``Other Settings`` field, as this is the only field the backend can access directly besides ID and Secret.
+All new associations will be correctly namespaced with the given slug. For existing associations the new
```

Comment:
why don't you add For existing associations... to the consequences section?

Reply:
The 'For existing...' part is supposed to explain how those existing associations are handled. So it seemed best have it there.